public_html/wp-content/mu-plugins/blocks/assets/src/organizers/organizers-select.js

```diff
@@ -28,45 +28,47 @@ class OrganizersSelect extends Component {
 		};
 
 		this.buildSelectOptions = this.buildSelectOptions.bind( this );
-		this.fetchSelectOptions( props );
 	}
 
-	fetchSelectOptions( props ) {
+	static getDerivedStateFromProps( props, state ) {
```

Comment:
🤔 I'm not sure. 

The post about moving away from the lifecycle methods [specifically discouraged `getDerivedStateFromProps()` as a replacement](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html). It sounds like the suggested approach for many cases was to refactor in order to remove anti-patterns, rather than just choosing a different lifecycle method.

Having said that, I don't feel like I understand things enough to know where this qualifies as one of the exceptions to that rule or not.

So, if you and @coreymckrill feel comfortable with it, and we don't notice any tangible problems, then it's probably fine to go ahead with this. We can always fix it later if we discover a problem.

Reply:
I found a way to refactor all of the blocks' select components to no longer use `getDerivedStateFromProps`. It seems to work and does not seem to cause excessive re-renders of the select options. Here's the commit for the Organizers block: 3df88a895deaefbb11a0958a5d340efd67f1a2e8

(I'm still working on refactoring the other blocks)

Reply:
With the latest commit on this PR, I have eliminated usage of `getDerivedStateFromProps`. It seemed like the right thing to do, but if anyone still wants to make the case for a valid usage, I'm open.

Reply:
@coreymckrill  This approach looks solid. There is not need to use a life cycle method if we can avoid it. :+1: